### PR TITLE
feat(experiments): UX refresh for the no-code toolbar.

### PR DESCRIPTION
**Pull request by Phanatic**
## Problem

@corywatilo and I did a design review of the no-code toolbar and found a lot of issues we could fix easily. 
Below is the list: 

 1. bug: h2, span is not selectable.
 2. bug: Cannot enter experiment name.
 3. ux: Control shouldn't ever change anything
 4. ux: Control shouldn't need to add any transforms.
 5. ux: Should not delete control
 6. ux: No visual frame to tell which variant, maybe create accordions.
 7. feat: should be able to edit the selector.
 8. feat: element selector should always
 12. Remove should be just the trash icon
 13. visualize section whenever its expanded.
 14. Control should always be first. 
15. Placeholder on accordion title.
16. Move all controls to the header.
17. Control blankslate
18. Open test variant when creating a new experiment.

### Before

nct_ux_stale

### After
nct_ux_refresh

#### After

👉 _Stay up-to-date with [PostHog coding conventions](https://posthog.com/docs/contribute/coding-conventions) for a smoother review._

## Does this work well for both Cloud and self-hosted?

Yes

## How did you test this code?

Manually, in local development

### CODE REVIEW

**Comment by jurajmajerik** [comment]
Do we need the Cancel/Save buttons here? I think a simple text input would do.

image

**Comment by jurajmajerik** [comment]
Clicking on the accordion items doesn't expand them.

https://github.com/user-attachments/assets/1c1496ae-d13f-4194-9e2f-5818121dc914

**Comment by jurajmajerik** [comment]
I'm not sure I like having the select button directly on the accordion header.

image

The purpose of the accordion header is only to collapse/uncollapse an accordion item. I don't think it's a good idea to mix it with the functionality to select an element.

There is no point selecting an element if a particular accordion item is collapsed, so in that case that functionality should be hidden.

My suggestion:
- Put the "Select element" button inside the accordion item content
- In the accordion header, render either:
	- The element selector (make sure it's truncated, these can get pretty long and break the UI)
	- Placeholder if no element selected: `No element selected` or similar

**Comment by jurajmajerik** [comment]
Change text to `Select element`

No need to be courteous on buttons :)) it should convey the meaning and take up as little space as possible

image

**Comment by jurajmajerik** [comment]
I would just use a ➕ icon here. The target icon is already used by the "Select element" button. Don't repeat icons for different buttons (generally)

image

**Comment by jurajmajerik** [comment]
image

We have a standardized success tag for the dark mode (LemonTag type="success") which looks like this:

image

**Comment by jurajmajerik** [comment]
I'd suggest a bit more elaborate copy here:

image

`You're viewing your page in its original state. The control variant serves as a baseline to compare all test variants against the original.`

**Comment by jurajmajerik** [comment]
Do not allow removing _all_ elements for a variant. At least one must always be present. Otherwise it's the same as `control`

image

**Comment by jurajmajerik** [comment]
Do not allow removing the last test variant - we always need the `control` and at least 1 test variant

image

**Comment by jurajmajerik** [comment]
Variant keys should be editable. Again, I suggesting putting the input field inside the accordion content, and only render the variant key on the accordion header (similar to https://github.com/PostHog/posthog/pull/25190#issuecomment-2374219297)

image

**Comment by jurajmajerik** [comment]
The main action button `Create experiment` should be positioned to the right, and `Cancel` to the left

image

**Comment by jurajmajerik** [comment]
There's a horizontal overflow here, rendering a scroll bar

https://github.com/user-attachments/assets/69d3b94c-81b6-4255-80cb-6cc50c168515

**Review by dmarticus — APPROVED** [review]

File: frontend/src/lib/actionUtils.ts [hunk 1]
```diff
@@ -3,8 +3,10 @@ import { cssEscape } from 'lib/utils/cssEscape'
 import { ElementType } from '~/types'
 
 // these plus any element with cursor:pointer will be click targets
-export const CLICK_TARGETS = ['a', 'button', 'input', 'select', 'textarea', 'label', 'h1']
+export const CLICK_TARGETS = ['a', 'button', 'input', 'select', 'textarea']
+export const EXPERIMENT_TARGETS = ['label', 'h1', 'h2', 'h3', 'h4', 'span', 'img']
```
Comment (Phanatic) [inline frontend/src/lib/actionUtils.ts:7]: @jurajmajerik or @corywatilo is this a good set of elements to run experiments on? I don't want to put in a `*` here and would rather limit the set.

Comment (jurajmajerik) [reply frontend/src/lib/actionUtils.ts:7]: I think `div` and `p` should be here as well? We can always extend this easily..

**Review by dmarticus — APPROVED** [review]
looks good to me, but I'll wait for the feedback from the rest of the team to settle, too.

File: frontend/src/lib/actionUtils.ts (hunk 1 above)
Comment (corywatilo) [reply frontend/src/lib/actionUtils.ts:7]: Tbh I think this is a huge limitation if we're deliberately specifying a select amount of tags. Going to lead to a lot of user frustration if they can't just target exactly what they're looking for.

Comment (jurajmajerik) [reply frontend/src/lib/actionUtils.ts:7]: oh, I've just realized there are semantic tags too..

**Review by jurajmajerik — APPROVED** [review]
LGTM!

**Pull request merged**